### ISSUE TITLE
Split on signing

### ISSUE THREAD

**Issue by Graem117** [caosdoar/SteelBattalionDriver#3]
Hey man, I be willing to split the cost if you want to get this signed. Would love to be able to use this with starwars squadrons.

**Comment by Graem117** [comment]
I also found this that might help

https://www.unknowncheats.me/forum/anti-cheat-bypass/323626-sign-driver-windows-x64-10-8-1-7-self-signed-certificate.html

**Comment by caosdoar** [comment]
@Graem117 you offer is very kind but I am not very keen in managing others people money, with something I am not sure I can follow through.

The driver needs to be approved by microsoft and I am not an experienced driver developer. So I am unaware of the amount of work/time it will require. But the code is available for anyone to give it a try.

About the link you share. That is for auto-signing the drivers, something that was possible with older versions of windows, but not with the most recent ones of Windows 10. Now it is required not only to be signed by the developer (what requires to pay for a certificate) but to be signed by Microsoft as well, and that means to go through a certification process with them.

You can find more information about the process (including the changes in Windows 10 I am refering) here:
https://docs.microsoft.com/en-us/windows-hardware/drivers/dashboard/get-a-code-signing-certificate

Regards,
Oscar.

**Comment by the-math-god** [comment]
Then please start a Kickstarter to get these drivers signed because I want to use this with star wars squadron but it won't start when in test sign mode

**Comment by naz-drala** [comment]
Looks like you can self sign the driver

http://woshub.com/how-to-sign-an-unsigned-driver-for-windows-7-x64/

**Comment by Graem117** [comment]
Not sure if that works with windows 10

**Comment by Graem117** [comment]
> Looks like you can self sign the driver
> 
> http://woshub.com/how-to-sign-an-unsigned-driver-for-windows-7-x64/

But if you can do it, please let me know how you did it.

**Comment by naz-drala** [comment]
Nope, looks like I was wrong. You still need to turn on test mode to use the driver even after going through all that. drat

**Comment by naz-drala** [comment]
NEVER MIOND ON THE NEVER MIND!!!

After following the directions on that site (the cert tools are here C:\WinDDK\7600.16385.1\bin\amd64, and inf2cat tool is here C:\WinDDK\7600.16385.1\bin\amd64)

Then you need to right click on the steelbattalion cat file and select properties. Go the the digital signature tab.
Select the signer and hit details.
Hit view cert
Then install it to local machine -> trusted root certification authorities and local machine -> trusted publishers
If you installed the driver with test on then uninstall it selecting delete driver then scan for hardware changes

![Capture](https://user-images.githubusercontent.com/10554532/97228706-af22b300-17ad-11eb-9fcf-5e95163bcbbd.JPG)

**Comment by Graem117** [comment]
Woah really!? Wtf! Okay so I am trying to understand, this works for windows 10 right? Cause I dont see winddk on the main C drive?

**Comment by naz-drala** [comment]
Yes it works in win 10
You need to install the sdk, that installs all the tools you need. http://www.microsoft.com/en-us/download/details.aspx?id=11800
The instructions for setting up the cert start at Tools for Signing Drivers

**Comment by Graem117** [comment]
Okay I got the tools, so I install the driver while test is on right?

**Comment by naz-drala** [comment]
No, install after.

**Comment by Graem117** [comment]
Okay I am in cert install and I cant seem to find the trusted root certification authorities and local machine
![Uploading image.png…]()

**Comment by naz-drala** [comment]
![Capture1](https://user-images.githubusercontent.com/10554532/97239024-1ac24b80-17c1-11eb-87cc-17a77808531a.JPG)
![Capture2](https://user-images.githubusercontent.com/10554532/97239030-1c8c0f00-17c1-11eb-9b96-f5f492f6dbc4.JPG)

**Comment by Graem117** [comment]
Oh I have to make the certificate right? I thought it was to use the existing one.

**Comment by naz-drala** [comment]
Yea, you need to go all the steps to make the cert and attach it to the cat file.

**Comment by Graem117** [comment]
Ooooh okay knew there was something there, okay wanted to be sure. Thanks!

**Comment by Graem117** [comment]
Okay certificate is installed, and now for the actual process itself, need to know the steps.

**Comment by Graem117** [comment]
So do I plug it in, device manager, install driver? or is there something else involved

**Comment by naz-drala** [comment]
If you have followed all the steps on that website I have linked and built the certifacte and used pnputil to install the certifacte to the driver. all you have to do is
Then you need to right click on the steelbattalion cat file and select properties. Go the the digital signature tab.
Select the signer and hit details.
Hit view cert
Then install it to local machine -> trusted root certification authorities and local machine -> trusted publishers
Now you can install the driver and it show up as signed.

**Comment by Graem117** [comment]
I am getting errors when I install the driver? I did all the steps and even check the certificate manger its there.

**Comment by Graem117** [comment]
![image](https://user-images.githubusercontent.com/67132870/97241607-ffa70a00-17c7-11eb-8cc4-0747fc222e14.png)

**Comment by Graem117** [comment]
Oh wait, let me try something

**Comment by Graem117** [comment]
Oh wait do I need to have the pedals attached?

**Comment by naz-drala** [comment]
No you do not need the pedals attached. 
You need to add both the cer file and the cat file added to the local machine trusted root and trusted publisher

**Comment by Graem117** [comment]
Mmmm I am gonna start over again to be sure
I feel like I am missing something

**Comment by Graem117** [comment]
Dont know if its too much to ask for a image step by step? or a youtube vid?

**Comment by Graem117** [comment]
Only thing I think is different for this
![image](https://user-images.githubusercontent.com/67132870/97243027-c1134e80-17cb-11eb-8f31-5011a337dfaa.png)

**Comment by Graem117** [comment]
I found the cert2spc in
![image](https://user-images.githubusercontent.com/67132870/97243054-d38d8800-17cb-11eb-829d-690ad0e85407.png)

**Comment by Graem117** [comment]
instead of the C: tools in the link you gave to me

**Comment by Graem117** [comment]
Okay I found the issue, When I create the Cat file and get it self signed it does not work even though it states it worked in the cmd

![image](https://user-images.githubusercontent.com/67132870/97243632-23b91a00-17cd-11eb-8294-fb9c1a21ae76.png)

**Comment by Graem117** [comment]
Okay certificate is okay, and driver does install, but still having issue
trying to decipher this what you type
"If you installed the driver with test on then uninstall it selecting delete driver then scan for hardware changes"

**Comment by Graem117** [comment]
So I installed it in test sign mode and it does work, when I turn test mode off it does not work anymore.

**Comment by Graem117** [comment]
> No you do not need the pedals attached.
> You need to add both the cer file and the cat file added to the local machine trusted root and trusted publisher

Hey can you do a favor form me, can you check the KMDF HID driver file details show me whats in it?

![image](https://user-images.githubusercontent.com/67132870/97245803-3bdf6800-17d2-11eb-8697-b1c3ce136f00.png)

**Comment by Graem117** [comment]
I am think its the version of the sdk is whats screwing me over, having trouble installing the 7.1 version, I got the 10 version of the sdk and I think thats whats screwing me up.

**Comment by Graem117** [comment]
Okay so I disabled secure boot, and did a number of way to install it. No success still getting this error. It only works when I have bcd test sign on. I have no idea where I am having trouble here.

![image](https://user-images.githubusercontent.com/67132870/97291765-ba62f680-1820-11eb-880d-7636d8c8cf45.png)

**Comment by Graem117** [comment]
The only thing I can think of is window version, or the order of how this driver is installed thats causing issue (Which the creator of this has stated above). I might take a look at this steel battalion setup and see if I can make that work.(https://sourceforge.net/projects/steel-batallion-64/)

Might give the signed tutorial one final shot before giving up.

**Comment by Graem117** [comment]
Main reason I think it could potentially still work is the comments of the self signed tutorial has people saying it works to even close to this current month. So there has to be something.

**Comment by Graem117** [comment]
Ya 3rd time no luck, so doing research, reason being this does not work is due to the kernel sys files. The driver installs fine, but ounce it loads the sys files it can detect there not signed and causes it to not work. I think it worked for you is cause you have a version of windows thats not been updated or its a 32 bit system.

**Comment by Graem117** [comment]
Got it to work using the steel batallion 64 method that I linked above
